### DISCUSSION TITLE
Issues with gofumpt and golangci-lint in LazyVim

### DISCUSSION THREAD

**Discussion by michael-burakov** [LazyVim/LazyVim#4362] [Q&A]
Hi,

I recently started to migrate to Neovim and Lua scripting, having previously used Vim for quite some time. I've been using LazyVim and am very impressed with its configurability.

As a Golang developer, I've been utilizing the `extras/lang/go` extras. I'm also a big fan of linters, especially `golangci-lint`, which I use to configure and run multiple linters in parallel.

I'm facing two main challenges related to `gofumpt` and `golangci-lint` in LazyVim:

1. Disabling `gofumpt`:
    - I prefer using other linters with autofix options, so I don't need `gofumpt`.
    - I've tried disabling `gofumpt` in `gopls` but without success.
    - How can I prevent `gofumpt` from being automatically installed by Mason as it is included in `extras/lang/go` default config?
    - I've tried to manually uninstall it in Mason, but it is automatically installed on next Neovim startup.

2. Autofix conflicts with `golangci-lint`:
    - I'm using `nvim-lint` to trigger `golangci-lint` on specific events: "BufWritePost", "BufReadPost", "InsertLeave" (as in example).
    - Autofix in `golangci-lint` is modifying source files directly.
    - This leads to outdated buffers and errors when saving/reading/reloading files.
    - What's the best approach to avoid these conflicts and ensure buffer updates after linting?

I'd appreciate any help, thanks!

**Comment by dpetka2001** [comment]
Regarding 1 you could do in your personal configuration
```lua
{
  "mason.nvim",
  opts = function(_, opts)
    opts.ensure_installed = vim.tbl_filter(function(c)
      return not vim.tbl_contains({ "gofumpt" }, c)
    end, opts.ensure_installed)
  end,
},
```
Regarding 2 I don't do any Go, so can't help there.

**Reply by michael-burakov** [reply]
Cool, thanks that's exactly what I was looking for to solve the first problem.
